packages/mgt-components/src/components/mgt-search-results/mgt-search-results.ts

```diff
@@ -0,0 +1,1220 @@
+   */
+  private renderPerson(result: SearchHit): HTMLTemplateResult {
+    let resource: any = result.resource as any;
+    return mgtHtml`
+      <div class="search-result">
+        <mgt-person 
+          view="fourLines" 
+          person-query=${resource.userPrincipalName} 
+          person-card="hover"
+          show-presence="true">
+        </mgt-person> 
+      </div>          
+      <fluent-divider></fluent-divider>
+    `;
+  }
+
+  /**
+   * Renders a bookmark entity
+   * @param result
+   * @returns
+   */
+  private renderBookmark(result: SearchHit): HTMLTemplateResult {
+    return this.renderAnswer(result, SvgIcon.DoubleBookmark);
+  }
+
+  /**
+   * Renders an acronym entity
+   * @param result
+   * @returns
+   */
+  private renderAcronym(result: SearchHit): HTMLTemplateResult {
+    return this.renderAnswer(result, SvgIcon.BookOpen);
+  }
+
+  /**
+   * Renders a qna entity
+   * @param result
+   * @returns
+   */
+  private renderQnA(result: SearchHit): HTMLTemplateResult {
+    return this.renderAnswer(result, SvgIcon.BookQuestion);
+  }
+
+  /**
+   * Renders an answer entity
+   * @param result
+   * @returns
+   */
+  private renderAnswer(result: SearchHit, icon: SvgIcon): HTMLTemplateResult {
+    let resource: any = result.resource as any;
+    return html`
+      <div class="search-result-grid search-result-answer">
+        <div class="search-result-icon">
+          ${getSvg(icon)}
+        </div>
+        <div class="search-result-content">
+          <div class="search-result-name">
+            <a href="${resource.webUrl}?Web=1" target="_blank">${resource.displayName}</a>
+          </div>
+          <div class="search-result-summary">${resource.description}</div>
+        </div>  
+      </div>          
+      <fluent-divider></fluent-divider>
+    `;
+  }
+
+  /**
+   * Renders any entity
+   * @param result
+   * @returns
+   */
+  private renderDefault(result: SearchHit): HTMLTemplateResult {
+    let resource: any = result.resource as any;
+    const resourceUrl = this.getResourceUrl(resource);
+    return html`
+      <div class="search-result-grid">
+        <div class="search-result-icon">
+          ${this.getResourceIcon(resource)}
+        </div>
+        <div class="search-result-content">
+          <div class="search-result-name">
+            ${
+              resourceUrl
+                ? html`
+                  <a href="${resourceUrl}?Web=1" target="_blank">${this.getResourceName(resource)}</a>
+                `
+                : html`
+                  ${this.getResourceName(resource)}
+                `
+            }            
+          </div>
+          <div class="search-result-summary" .innerHTML="${this.getResultSummary(result)}"></div>
+        </div>  
+      </div>          
+      <fluent-divider></fluent-divider>
+    `;
+  }
+
+  /**
+   * Gets default resource URLs
+   * @param resource
+   * @returns
+   */
+  private getResourceUrl(resource: any): string {
```

Comment:
It actually doesn't. It's the contained `resource` from the search results. As they can be of X types (`driveItem`, `message`, `qna`, etc.) there is nothing shared amongst them... It's a real, real `any` here.

Reply:
So, it's a union type across a number of types that we know of.

We can also cheat a little here and declare a type to cast to instead of any. e.g

```typescript
type SearchResource = {
  displayName?: string;
  name?: string;
  subject?: string;
  url?: string;
  webLink?: string;
  webUrl?: string;
}

```